DRAFT blog  - Learn MicroProfile 5.0 and Jakarta EE 9.1 from our newly enhanced guides (#2180)
OpenLiberty/blogs: posts/2022-04-12-guides-updated-mp5-jakartaee9.adoc

---
layout: post
title: "Learn MicroProfile 5.0 and Jakarta EE 9.1 from our newly enhanced guides"
# Do NOT change the categories section
categories: blog
author_picture: https://avatars3.githubusercontent.com/gkwan-ibm
author_github: https://github.com/gkwan-ibm
seo-title: Learn MicroProfile 5.0 and Jakarta EE 9.1 from Open Liberty guides
seo-description: Learn how to develop microservices on Open Liberty with MicroProfile 5.0 and Jakarta EE 9.1 from the guides at the openliberty.io site.
blog_description: "Learn how to develop microservices on Open Liberty with MicroProfile 5.0 and Jakarta EE 9.1 from our newly enhanced guides at the openliberty.io site."

> open-graph-image: https://openliberty.io/img/twitter_card.jpg

Reviewer: ```suggestion
open-graph-image: https://openliberty.io/img/blog/blog-2168-SNL-progress-bar.png
```

---
= Learn MicroProfile 5.0 and Jakarta EE 9.1 from our newly enhanced guides
Gilbert Kwan <https://github.com/gkwan-ibm>
:imagesdir: /
:url-prefix:
:url-about: /
//Blank line here is necessary before starting the body of the post.

> To learn how to develop microservices on Open Liberty with MicroProfile 5.0 and Jakarta EE 9.1, you can try out our updated guides. Excitingly, you can run our guides in a hosted environment with enhanced interfaces, without having to worry about local setup and prerequisites. Beside the Maven project pom.xml and Liberty configuration server.xml files are updated, the Java classes are using the jakarta package namespace instead of javax.

Reviewer: ```suggestion
To learn how to develop microservices on Open Liberty with MicroProfile 5.0 and Jakarta EE 9.1, you can try out our updated guides. Excitingly, you can run our guides in a hosted environment with enhanced interfaces, without having to worry about local setup and prerequisites. We recently updated 31 of our guides for compatibility with MicroProfile 5.0 and Jakarta EE 9. The Maven project `pom.xml` and Liberty configuration `server.xml` files in these guides are updated to use new Jakarta EE 9.1 and MicroProfile 5.0 features. Also, all Java classes now use the `jakarta` package namespace instead of `javax`, as shown in the following example.
```

image::/img/blog/blog-2168-SNL-java-class.png[Java class sample,width=40%,align="center"]

> To learn more about what are changed in Jakarta EE 9.1, visit the link:https://openliberty.io/docs/latest/jakarta-ee9-feature-updates.html[Jakarta EE 9.1 feature updates^] documentation. To learn the different between MicroProfile 5.0 and 4.1, visit the link:https://openliberty.io/docs/latest/mp-41-50-diff.html[Differences between MicroProfile 5.0 and 4.1^] documentation.

Reviewer: ```suggestion
To learn more about what changed in Jakarta EE 9.1, see the link:https://openliberty.io/docs/latest/jakarta-ee9-feature-updates.html[Jakarta EE 9.1 feature updates^] documentation. To learn about the differences between MicroProfile 5.0 and 4.1, see the link:https://openliberty.io/docs/latest/mp-41-50-diff.html[Differences between MicroProfile 5.0 and 4.1^] documentation.
```

> == What new enhancements in the cloud-hosted guides?

Reviewer: ```suggestion
== What are the new enhancements in the cloud-hosted guides?
```

> The cloud-hosted guides are enhanced with new styles. No matter you experienced the cloud-hosted guides or not. We recommend you to experience them how simply to learn the MicroProfile and Jakarta EE features by the following new enhancements in the cloud-hosted guides. You can access them by the links listed on the bottom of this blog or visit our link:{url-prefix}/guides/[guides], and clicking the *Run in cloud* button, as shown in the following image:

Reviewer: ```suggestion
The cloud-hosted guides are enhanced with a new look and updated features. Whether or not you've tried them yet, now is the time to experience how simply you can learn the MicroProfile and Jakarta EE features with our newly enhancemed cloud-hosted guides. 

You can access any cloud hosted guide by clicking one of the links at the end of this blog or by selecting one from link:{url-prefix}/guides/[our full collection of guides]. After you choose a guide, click the *Run in cloud* button, as shown in the following image:
```

image::/img/blog/blog-2168-guide.png[Guide sample,width=70%,align="center"]

> 

Reviewer: ```suggestion

We've enhanced these guides to make them easier to navigate and to simplify running commands and managing files. 
```

> Following screenshot shows the new look and you can navigate to different sections of the guide by the�image:/img/blog/blog-2168-SNL-toc-icon.png[] *Table of Contents* menu:

Reviewer: ```suggestion
The following screenshot shows the new look for our cloud hosted guides. You can navigate to different sections of the guide by clicking the �image:/img/blog/blog-2168-SNL-toc-icon.png[table of contents icon] *Table of Contents* menu:
```

image::/img/blog/blog-2168-SNL-TOC.png[Guide sample,width=80%,align="center"]

> Or, you can navigate to different sections of the guide through the blue progress bar under the Table of Contents menu:

Reviewer: ```suggestion
You can also navigate through the guide by clicking the blue progress bar after the Table of Contents menu:
```

image::/img/blog/blog-2168-SNL-progress-bar.png[Guide sample,width=50%,align="center"]

> Instead of copying the commands and pasting them on terminal, now, you can run the commands on an active terminal by simply clicking on the Execute `>_` icon.

Reviewer: ```suggestion
Instead of copying and pasting the commands in terminal,  you can now run the commands on an active terminal by simply clicking on the Execute `>_` icon in the code window.
```

image::/img/blog/blog-2168-SNL-exec.png[Guide sample,width=50%,align="center"]

> You can directly open file by clicking the *Open ... in IDE* button, instead of browsing file from the *File* > *Open* menu.

Reviewer: ```suggestion
You can now directly open a file by clicking the *Open ... in IDE* button, instead of browsing to it from the *File* > *Open* menu.
```

image::/img/blog/blog-2168-SNL-open-file.png[Guide sample,width=50%,align="center"]

> Also, you can simply launch application by clicking on the launch button. The application URL will be opened on a new browser tab.

Reviewer: ```suggestion
Also, you can  launch an application simply by clicking on the launch button. The application URL opens on a new browser tab.
```

* link:{url-prefix}/guides/microprofile-opentracing-jaeger.html[Enabling distributed tracing in microservices with Jaeger^]
* link:{url-prefix}/guides/getting-started.html[Getting started with Open Liberty^]
* link:{url-prefix}/guides/cdi-intro.html[Injecting dependencies into microservices^]
* link:{url-prefix}/guides/istio-intro.html[Managing microservice traffic using Istio^]
* link:{url-prefix}/guides/mongodb-intro.html[Persisting data with MongoDB^]
* link:{url-prefix}/guides/microprofile-metrics.html[Providing metrics from a microservice^]
* link:{url-prefix}/guides/security-intro.html[Securing a web application^]
* link:{url-prefix}/guides/microprofile-jwt.html[Securing microservices with JSON Web Tokens^]
* link:{url-prefix}/guides/contract-testing.html[Testing microservices with consumer-driven contracts^]
* link:{url-prefix}/guides/arquillian-managed.html[Testing microservices with the Arquillian managed container^]
* link:{url-prefix}/guides/docker.html[Using Docker containers to develop microservices^]
* link:{url-prefix}/guides/bean-validation.html[Validating constraints with microservices^]

> == Your feedback is welcomed

Reviewer: ```suggestion
== We welcome your feedback
```
